packages/backend/src/services/SpaceService/SpaceService.ts

```diff
@@ -100,18 +109,24 @@ export class SpaceService {
         spaceUuid: string,
         updateSpace: UpdateSpace,
     ): Promise<Space> {
-        const space = await this.spaceModel.get(spaceUuid);
+        const space = await this.spaceModel.getFullSpace(spaceUuid);
         if (
             user.ability.cannot(
                 'manage',
                 subject('Space', {
                     organizationUuid: space.organizationUuid,
                     projectUuid: space.projectUuid,
                 }),
-            )
+            ) ||
+            !hasSpaceAccess(space, user.userUuid)
         ) {
             throw new ForbiddenError();
         }
+
+        if (space.isPrivate !== updateSpace.isPrivate) {
+            // Switching public and private spaces switches between their defaults
+            await this.spaceModel.clearSpaceAccess(spaceUuid);
```

Comment:
Should we keep access for the user that just changed the space ? otherwise no one will have access to it when updated to private.

Reply:
damn, yes you're right :facepalm:

Reply:
Fixed and tested
![Screenshot from 2022-11-11 16-33-32](https://user-images.githubusercontent.com/1983672/201374328-d597d3ee-63ce-4011-8304-001db1915085.png)
![Screenshot from 2022-11-11 16-37-38](https://user-images.githubusercontent.com/1983672/201374352-e710648f-fbd2-41a0-9017-018b79a12031.png)


![Screenshot from 2022-11-11 16-39-21](https://user-images.githubusercontent.com/1983672/201374366-7d714a2e-e7d5-4299-9927-e5377ee9a971.png)